src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Icu.cs

```diff
@@ -193,6 +193,15 @@ private unsafe int IndexOfOrdinalIgnoreCaseHelper(ReadOnlySpan<char> source, Rea
                         throw new Exception((string)ex_result);
                     return result;
                 }
+#elif TARGET_OSX || TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
+                if (GlobalizationMode.Hybrid)
+                {
+                    NSRange result = Interop.Globalization.IndexOfNative(m_name, m_name.Length, b, target.Length, a, source.Length, options, fromBeginning);
+                    if (matchLengthPtr == null)
+                        matchLengthPtr = &result.Length;
+                    *matchLengthPtr = result.Length;
```

Comment:
I'm not sure if we should be reassigning `matchLengthPtr` if its `null`.
We can probably match
```
if (matchLengthPtr != null)
    *matchLengthPtr = result.Length;
```

Reply:
Yes you are right, updated.